{"nbformat":4,"nbformat_minor":0,"metadata":{"colab":{"name":"MOFA on chronic lymphocytic leukemia.ipynb","provenance":[],"authorship_tag":"ABX9TyPLT34S9j0pkGkkBxNtiiv1"},"kernelspec":{"name":"ir","display_name":"R"},"language_info":{"name":"R"}},"cells":[{"cell_type":"code","metadata":{"colab":{"base_uri":"https://localhost:8080/"},"id":"5NSAWm5j_xbo","executionInfo":{"status":"ok","timestamp":1627457912620,"user_tz":-480,"elapsed":764543,"user":{"displayName":"Indrik Wijaya","photoUrl":"","userId":"00208067985282844199"}},"outputId":"fae11c25-c49a-4d89-dd64-e085d0285c9b"},"source":["if (!requireNamespace(\"BiocManager\", quietly = TRUE))\n","    install.packages(\"BiocManager\")\n","\n","BiocManager::install(\"DESeq2\")\n","BiocManager::install(\"MOFA\")\n","BiocManager::install(\"MultiAssayExperiment\")\n","BiocManager::install(\"limma\")\n","BiocManager::install(\"GEOquery\")"],"execution_count":1,"outputs":[{"output_type":"stream","text":["Installing package into ‘/usr/local/lib/R/site-library’\n","(as ‘lib’ is unspecified)\n","\n","'getOption(\"repos\")' replaces Bioconductor standard repositories, see\n","'?repositories' for details\n","\n","replacement repositories:\n","    CRAN: https://cran.rstudio.com\n","\n","\n","Bioconductor version 3.13 (BiocManager 1.30.16), R 4.1.0 (2021-05-18)\n","\n","Installing package(s) 'BiocVersion', 'DESeq2'\n","\n","also installing the dependencies ‘bit’, ‘bitops’, ‘formatR’, ‘bit64’, ‘plogr’, ‘png’, ‘Biostrings’, ‘RCurl’, ‘GenomeInfoDbData’, ‘zlibbioc’, ‘matrixStats’, ‘lambda.r’, ‘futile.options’, ‘RSQLite’, ‘KEGGREST’, ‘XML’, ‘xtable’, ‘GenomeInfoDb’, ‘XVector’, ‘MatrixGenerics’, ‘DelayedArray’, ‘futile.logger’, ‘snow’, ‘AnnotationDbi’, ‘annotate’, ‘S4Vectors’, ‘IRanges’, ‘GenomicRanges’, ‘SummarizedExperiment’, ‘BiocGenerics’, ‘Biobase’, ‘BiocParallel’, ‘genefilter’, ‘locfit’, ‘geneplotter’, ‘RcppArmadillo’\n","\n","\n","Old packages: 'blob', 'broom', 'cli', 'credentials', 'googledrive',\n","  'googlesheets4', 'openssl', 'readr', 'rvest', 'stringi', 'tibble', 'utf8'\n","\n","'getOption(\"repos\")' replaces Bioconductor standard repositories, see\n","'?repositories' for details\n","\n","replacement repositories:\n","    CRAN: https://cran.rstudio.com\n","\n","\n","Bioconductor version 3.13 (BiocManager 1.30.16), R 4.1.0 (2021-05-18)\n","\n","Installing package(s) 'MOFA'\n","\n","Warning message in .inet_warning(msg):\n","“package ‘MOFA’ is not available for Bioconductor version '3.13'\n","\n","A version of this package for your version of R might be available elsewhere,\n","see the ideas at\n","https://cran.r-project.org/doc/manuals/r-patched/R-admin.html#Installing-packages”\n","Old packages: 'blob', 'broom', 'cli', 'credentials', 'googledrive',\n","  'googlesheets4', 'openssl', 'readr', 'rvest', 'stringi', 'tibble', 'utf8'\n","\n","'getOption(\"repos\")' replaces Bioconductor standard repositories, see\n","'?repositories' for details\n","\n","replacement repositories:\n","    CRAN: https://cran.rstudio.com\n","\n","\n","Bioconductor version 3.13 (BiocManager 1.30.16), R 4.1.0 (2021-05-18)\n","\n","Installing package(s) 'MultiAssayExperiment'\n","\n","Old packages: 'blob', 'broom', 'cli', 'credentials', 'googledrive',\n","  'googlesheets4', 'openssl', 'readr', 'rvest', 'stringi', 'tibble', 'utf8'\n","\n","'getOption(\"repos\")' replaces Bioconductor standard repositories, see\n","'?repositories' for details\n","\n","replacement repositories:\n","    CRAN: https://cran.rstudio.com\n","\n","\n","Bioconductor version 3.13 (BiocManager 1.30.16), R 4.1.0 (2021-05-18)\n","\n","Installing package(s) 'limma'\n","\n","Old packages: 'blob', 'broom', 'cli', 'credentials', 'googledrive',\n","  'googlesheets4', 'openssl', 'readr', 'rvest', 'stringi', 'tibble', 'utf8'\n","\n","'getOption(\"repos\")' replaces Bioconductor standard repositories, see\n","'?repositories' for details\n","\n","replacement repositories:\n","    CRAN: https://cran.rstudio.com\n","\n","\n","Bioconductor version 3.13 (BiocManager 1.30.16), R 4.1.0 (2021-05-18)\n","\n","Installing package(s) 'GEOquery'\n","\n","Old packages: 'blob', 'broom', 'cli', 'credentials', 'googledrive',\n","  'googlesheets4', 'openssl', 'readr', 'rvest', 'stringi', 'tibble', 'utf8'\n","\n"],"name":"stderr"}]},{"cell_type":"code","metadata":{"id":"turASPSaCiP4","colab":{"base_uri":"https://localhost:8080/"},"executionInfo":{"status":"ok","timestamp":1627458880232,"user_tz":-480,"elapsed":967637,"user":{"displayName":"Indrik Wijaya","photoUrl":"","userId":"00208067985282844199"}},"outputId":"e5690779-cbb7-48fa-e683-e1db44aa595c"},"source":["install.packages(c('reticulate', 'survival', 'survminer', 'maxstat',\n","'gridExtra', 'car', 'cowplot', 'egg', 'ggrepel', 'ggbeeswarm', 'grImport',\n","'pheatmap', 'UpSetR', 'glmnet', 'tidygraph', 'igraph', 'ggraph',\n","'scattermore', 'RColorBrewer', 'circlize', 'lubridate'))"],"execution_count":2,"outputs":[{"output_type":"stream","text":["Installing packages into ‘/usr/local/lib/R/site-library’\n","(as ‘lib’ is unspecified)\n","\n","Warning message:\n","“packages ‘mofaCLL’, ‘sva’, ‘fgsea’, ‘ComplexHeatmap’, ‘CATALYST’, ‘diffcyt’, ‘genefilter’ are not available for this version of R\n","\n","Versions of these packages for your version of R might be available elsewhere,\n","see the ideas at\n","https://cran.r-project.org/doc/manuals/r-patched/R-admin.html#Installing-packages”\n","also installing the dependencies ‘corrplot’, ‘jpeg’, ‘ggsci’, ‘ggsignif’, ‘polynom’, ‘rstatix’, ‘KMsurv’, ‘zoo’, ‘km.ci’, ‘gridtext’, ‘numDeriv’, ‘SparseM’, ‘MatrixModels’, ‘conquer’, ‘sp’, ‘openxlsx’, ‘minqa’, ‘nloptr’, ‘RcppEigen’, ‘iterators’, ‘tweenr’, ‘polyclip’, ‘ggpubr’, ‘survMisc’, ‘ggtext’, ‘exactRankTests’, ‘mvtnorm’, ‘carData’, ‘abind’, ‘pbkrtest’, ‘quantreg’, ‘maptools’, ‘rio’, ‘lme4’, ‘beeswarm’, ‘vipor’, ‘plyr’, ‘foreach’, ‘shape’, ‘ggforce’, ‘viridis’, ‘graphlayouts’, ‘GlobalOptions’\n","\n","\n"],"name":"stderr"}]},{"cell_type":"code","metadata":{"colab":{"base_uri":"https://localhost:8080/"},"id":"EbO8LCGWDb-A","executionInfo":{"status":"ok","timestamp":1627461636862,"user_tz":-480,"elapsed":2755603,"user":{"displayName":"Indrik Wijaya","photoUrl":"","userId":"00208067985282844199"}},"outputId":"4d36c5d8-b09c-447c-9ce0-d25ebf201e0f"},"source":["devtools::install_github(\"bioFAM/MOFA\", build_opts = c(\"--no-resave-data\"))\n","BiocManager::install(\"sva\")\n","BiocManager::install(\"fgsea\")\n","BiocManager::install(\"ComplexHeatmap\")\n","BiocManager::install(\"CATALYST\")\n","BiocManager::install(\"diffcyt\")\n","BiocManager::install(\"genefilter\")\n","\n"],"execution_count":3,"outputs":[{"output_type":"stream","text":["Downloading GitHub repo bioFAM/MOFA@HEAD\n","\n"],"name":"stderr"},{"output_type":"stream","text":["utf8         (1.2.1 -> 1.2.2 ) [CRAN]\n","cli          (3.0.0 -> 3.0.1 ) [CRAN]\n","tibble       (3.1.2 -> 3.1.3 ) [CRAN]\n","stringi      (1.7.2 -> 1.7.3 ) [CRAN]\n","Rhdf5lib     (NA    -> 1.14.2) [CRAN]\n","reshape      (NA    -> 0.8.8 ) [CRAN]\n","rhdf5filters (NA    -> 1.4.0 ) [CRAN]\n","doParallel   (NA    -> 1.0.16) [CRAN]\n","GGally       (NA    -> 2.1.2 ) [CRAN]\n","reshape2     (NA    -> 1.4.4 ) [CRAN]\n","rhdf5        (NA    -> 2.36.0) [CRAN]\n"],"name":"stdout"},{"output_type":"stream","text":["Installing 11 packages: utf8, cli, tibble, stringi, Rhdf5lib, reshape, rhdf5filters, doParallel, GGally, reshape2, rhdf5\n","\n","Installing packages into ‘/usr/local/lib/R/site-library’\n","(as ‘lib’ is unspecified)\n","\n"],"name":"stderr"},{"output_type":"stream","text":["\u001b[32m✔\u001b[39m  \u001b[90mchecking for file ‘/tmp/Rtmp5Gbinz/remotes3d35a41818/bioFAM-MOFA-bf7df2f/DESCRIPTION’\u001b[39m\u001b[36m\u001b[39m\n","\u001b[90m─\u001b[39m\u001b[90m  \u001b[39m\u001b[90mpreparing ‘MOFA’:\u001b[39m\u001b[36m\u001b[39m\n","\u001b[32m✔\u001b[39m  \u001b[90mchecking DESCRIPTION meta-information\u001b[39m\u001b[36m\u001b[39m\n","\u001b[90m─\u001b[39m\u001b[90m  \u001b[39m\u001b[90mchecking for LF line-endings in source and make files and shell scripts\u001b[39m\u001b[36m\u001b[39m\n","\u001b[90m─\u001b[39m\u001b[90m  \u001b[39m\u001b[90mchecking for empty or unneeded directories\u001b[39m\u001b[36m\u001b[39m\n","   Omitted ‘LazyData’ from DESCRIPTION\n","\u001b[90m─\u001b[39m\u001b[90m  \u001b[39m\u001b[90mbuilding ‘MOFA_1.3.1.tar.gz’\u001b[39m\u001b[36m\u001b[39m\n","   \n"],"name":"stdout"},{"output_type":"stream","text":["Installing package into ‘/usr/local/lib/R/site-library’\n","(as ‘lib’ is unspecified)\n","\n","'getOption(\"repos\")' replaces Bioconductor standard repositories, see\n","'?repositories' for details\n","\n","replacement repositories:\n","    CRAN: https://cran.rstudio.com\n","\n","\n","Bioconductor version 3.13 (BiocManager 1.30.16), R 4.1.0 (2021-05-18)\n","\n","Installing package(s) 'sva'\n","\n","also installing the dependency ‘edgeR’\n","\n","\n","Old packages: 'blob', 'broom', 'cli', 'credentials', 'googledrive',\n","  'googlesheets4', 'openssl', 'readr', 'rvest', 'stringi', 'tibble', 'utf8'\n","\n","'getOption(\"repos\")' replaces Bioconductor standard repositories, see\n","'?repositories' for details\n","\n","replacement repositories:\n","    CRAN: https://cran.rstudio.com\n","\n","\n","Bioconductor version 3.13 (BiocManager 1.30.16), R 4.1.0 (2021-05-18)\n","\n","Installing package(s) 'fgsea'\n","\n","also installing the dependency ‘fastmatch’\n","\n","\n","Old packages: 'blob', 'broom', 'cli', 'credentials', 'googledrive',\n","  'googlesheets4', 'openssl', 'readr', 'rvest', 'stringi', 'tibble', 'utf8'\n","\n","'getOption(\"repos\")' replaces Bioconductor standard repositories, see\n","'?repositories' for details\n","\n","replacement repositories:\n","    CRAN: https://cran.rstudio.com\n","\n","\n","Bioconductor version 3.13 (BiocManager 1.30.16), R 4.1.0 (2021-05-18)\n","\n","Installing package(s) 'ComplexHeatmap'\n","\n","also installing the dependencies ‘rjson’, ‘GetoptLong’, ‘clue’, ‘Cairo’\n","\n","\n","Warning message in .inet_warning(msg):\n","“installation of package ‘Cairo’ had non-zero exit status”\n","Warning message in .inet_warning(msg):\n","“installation of package ‘ComplexHeatmap’ had non-zero exit status”\n","Old packages: 'blob', 'broom', 'cli', 'credentials', 'googledrive',\n","  'googlesheets4', 'openssl', 'readr', 'rvest', 'stringi', 'tibble', 'utf8'\n","\n","'getOption(\"repos\")' replaces Bioconductor standard repositories, see\n","'?repositories' for details\n","\n","replacement repositories:\n","    CRAN: https://cran.rstudio.com\n","\n","\n","Bioconductor version 3.13 (BiocManager 1.30.16), R 4.1.0 (2021-05-18)\n","\n","Installing package(s) 'CATALYST'\n","\n","also installing the dependencies ‘ash’, ‘hdrcde’, ‘rainbow’, ‘fds’, ‘tmvnsim’, ‘misc3d’, ‘DEoptimR’, ‘IDPmisc’, ‘fda’, ‘ellipse’, ‘mnormt’, ‘FNN’, ‘kernlab’, ‘mclust’, ‘multicool’, ‘plot3D’, ‘pracma’, ‘robustbase’, ‘pcaPP’, ‘R.oo’, ‘R.methodsS3’, ‘flowViz’, ‘flowStats’, ‘flowClust’, ‘ks’, ‘rrcov’, ‘R.utils’, ‘hexbin’, ‘TH.data’, ‘sandwich’, ‘RcppParallel’, ‘openCyto’, ‘RBGL’, ‘Rgraphviz’, ‘graph’, ‘ggcyto’, ‘corpcor’, ‘RUnit’, ‘latticeExtra’, ‘aws.s3’, ‘aws.signature’, ‘ncdfFlow’, ‘sparseMatrixStats’, ‘RcppHNSW’, ‘ScaledMatrix’, ‘irlba’, ‘rsvd’, ‘Cairo’, ‘ALL’, ‘gtools’, ‘multcomp’, ‘plotrix’, ‘cytolib’, ‘RProtoBufLib’, ‘colorRamps’, ‘CytoML’, ‘flowWorkspace’, ‘ggnewscale’, ‘ggpointdensity’, ‘scuttle’, ‘DelayedMatrixStats’, ‘beachmat’, ‘BiocNeighbors’, ‘BiocSingular’, ‘SingleCellExperiment’, ‘ComplexHeatmap’, ‘ConsensusClusterPlus’, ‘drc’, ‘flowCore’, ‘FlowSOM’, ‘ggridges’, ‘nnls’, ‘Rtsne’, ‘scater’\n","\n","\n","Warning message in .inet_warning(msg):\n","“installation of package ‘Cairo’ had non-zero exit status”\n","Warning message in .inet_warning(msg):\n","“installation of package ‘ComplexHeatmap’ had non-zero exit status”\n","Warning message in .inet_warning(msg):\n","“installation of package ‘CATALYST’ had non-zero exit status”\n","Old packages: 'blob', 'broom', 'cli', 'credentials', 'googledrive',\n","  'googlesheets4', 'openssl', 'readr', 'rvest', 'stringi', 'tibble', 'utf8'\n","\n","'getOption(\"repos\")' replaces Bioconductor standard repositories, see\n","'?repositories' for details\n","\n","replacement repositories:\n","    CRAN: https://cran.rstudio.com\n","\n","\n","Bioconductor version 3.13 (BiocManager 1.30.16), R 4.1.0 (2021-05-18)\n","\n","Installing package(s) 'diffcyt'\n","\n","also installing the dependencies ‘Cairo’, ‘ComplexHeatmap’\n","\n","\n","Warning message in .inet_warning(msg):\n","“installation of package ‘Cairo’ had non-zero exit status”\n","Warning message in .inet_warning(msg):\n","“installation of package ‘ComplexHeatmap’ had non-zero exit status”\n","Warning message in .inet_warning(msg):\n","“installation of package ‘diffcyt’ had non-zero exit status”\n","Old packages: 'blob', 'broom', 'cli', 'credentials', 'googledrive',\n","  'googlesheets4', 'openssl', 'readr', 'rvest', 'stringi', 'tibble', 'utf8'\n","\n","'getOption(\"repos\")' replaces Bioconductor standard repositories, see\n","'?repositories' for details\n","\n","replacement repositories:\n","    CRAN: https://cran.rstudio.com\n","\n","\n","Bioconductor version 3.13 (BiocManager 1.30.16), R 4.1.0 (2021-05-18)\n","\n","Warning message:\n","“package(s) not installed when version(s) same as current; use `force = TRUE` to\n","  re-install: 'genefilter'”\n","Old packages: 'blob', 'broom', 'cli', 'credentials', 'googledrive',\n","  'googlesheets4', 'openssl', 'readr', 'rvest', 'stringi', 'tibble', 'utf8'\n","\n"],"name":"stderr"}]},{"cell_type":"code","metadata":{"colab":{"base_uri":"https://localhost:8080/"},"id":"zz1zA207Rsgq","executionInfo":{"status":"ok","timestamp":1627461724098,"user_tz":-480,"elapsed":1137,"user":{"displayName":"Indrik Wijaya","photoUrl":"","userId":"00208067985282844199"}},"outputId":"726549ff-7cfa-499f-ac60-aa87da41c802"},"source":["install.packages('mofaCLL')"],"execution_count":7,"outputs":[{"output_type":"stream","text":["Installing package into ‘/usr/local/lib/R/site-library’\n","(as ‘lib’ is unspecified)\n","\n","Warning message:\n","“package ‘mofaCLL’ is not available for this version of R\n","\n","A version of this package for your version of R might be available elsewhere,\n","see the ideas at\n","https://cran.r-project.org/doc/manuals/r-patched/R-admin.html#Installing-packages”\n"],"name":"stderr"}]},{"cell_type":"code","metadata":{"colab":{"base_uri":"https://localhost:8080/"},"id":"-6-ZYicdCe2W","executionInfo":{"status":"ok","timestamp":1627461828303,"user_tz":-480,"elapsed":1503,"user":{"displayName":"Indrik Wijaya","photoUrl":"","userId":"00208067985282844199"}},"outputId":"a0583e8c-04a2-459b-da4a-92b82329e776"},"source":["knitr::opts_chunk$set(warning = FALSE, message = FALSE)\n","#library(mofaCLL)\n","library(reticulate)\n","library(DESeq2)\n","library(sva)\n","library(MultiAssayExperiment)\n","library(MOFA)\n","library(survival)\n","library(survminer)\n","library(maxstat)\n","library(gridExtra)\n","library(car)\n","library(cowplot)\n","library(egg)\n","library(ggrepel)\n","library(ggbeeswarm)\n","library(grImport)\n","library(fgsea)\n","library(limma)\n","library(GEOquery)\n","library(pheatmap)\n","library(UpSetR)\n","library(glmnet)\n","library(tidygraph)\n","library(igraph)\n","library(ggraph)\n","#library(ComplexHeatmap)\n","library(scattermore)\n","#library(CATALYST)\n","library(RColorBrewer)\n","library(circlize)\n","#library(diffcyt)\n","library(genefilter)\n","library(lubridate)\n","library(tidyverse)"],"execution_count":11,"outputs":[{"output_type":"stream","text":["Warning message in system(\"timedatectl\", intern = TRUE):\n","“running command 'timedatectl' had status 1”\n","\n","Attaching package: ‘lubridate’\n","\n","\n","The following objects are masked from ‘package:igraph’:\n","\n","    %--%, union\n","\n","\n","The following object is masked from ‘package:cowplot’:\n","\n","    stamp\n","\n","\n","The following objects are masked from ‘package:GenomicRanges’:\n","\n","    intersect, setdiff, union\n","\n","\n","The following object is masked from ‘package:GenomeInfoDb’:\n","\n","    intersect\n","\n","\n","The following objects are masked from ‘package:IRanges’:\n","\n","    %within%, intersect, setdiff, union\n","\n","\n","The following objects are masked from ‘package:S4Vectors’:\n","\n","    intersect, second, second<-, setdiff, union\n","\n","\n","The following objects are masked from ‘package:BiocGenerics’:\n","\n","    intersect, setdiff, union\n","\n","\n","The following objects are masked from ‘package:base’:\n","\n","    date, intersect, setdiff, union\n","\n","\n","── \u001b[1mAttaching packages\u001b[22m ─────────────────────────────────────── tidyverse 1.3.1 ──\n","\n","\u001b[32m✔\u001b[39m \u001b[34mtibble \u001b[39m 3.1.3     \u001b[32m✔\u001b[39m \u001b[34mdplyr  \u001b[39m 1.0.7\n","\u001b[32m✔\u001b[39m \u001b[34mtidyr  \u001b[39m 1.1.3     \u001b[32m✔\u001b[39m \u001b[34mstringr\u001b[39m 1.4.0\n","\u001b[32m✔\u001b[39m \u001b[34mreadr  \u001b[39m 1.4.0     \u001b[32m✔\u001b[39m \u001b[34mforcats\u001b[39m 0.5.1\n","\u001b[32m✔\u001b[39m \u001b[34mpurrr  \u001b[39m 0.3.4     \n","\n","── \u001b[1mConflicts\u001b[22m ────────────────────────────────────────── tidyverse_conflicts() ──\n","\u001b[31m✖\u001b[39m \u001b[34mlubridate\u001b[39m::\u001b[32m%--%()\u001b[39m        masks \u001b[34migraph\u001b[39m::%--%()\n","\u001b[31m✖\u001b[39m \u001b[34mlubridate\u001b[39m::\u001b[32m%within%()\u001b[39m    masks \u001b[34mIRanges\u001b[39m::%within%()\n","\u001b[31m✖\u001b[39m \u001b[34mdplyr\u001b[39m::\u001b[32mas_data_frame()\u001b[39m   masks \u001b[34mtibble\u001b[39m::as_data_frame(), \u001b[34migraph\u001b[39m::as_data_frame()\n","\u001b[31m✖\u001b[39m \u001b[34mlubridate\u001b[39m::\u001b[32mas.difftime()\u001b[39m masks \u001b[34mbase\u001b[39m::as.difftime()\n","\u001b[31m✖\u001b[39m \u001b[34mdplyr\u001b[39m::\u001b[32mcollapse()\u001b[39m        masks \u001b[34mnlme\u001b[39m::collapse(), \u001b[34mIRanges\u001b[39m::collapse()\n","\u001b[31m✖\u001b[39m \u001b[34mdplyr\u001b[39m::\u001b[32mcombine()\u001b[39m         masks \u001b[34mgridExtra\u001b[39m::combine(), \u001b[34mBiobase\u001b[39m::combine(), \u001b[34mBiocGenerics\u001b[39m::combine()\n","\u001b[31m✖\u001b[39m \u001b[34mpurrr\u001b[39m::\u001b[32mcompose()\u001b[39m         masks \u001b[34migraph\u001b[39m::compose()\n","\u001b[31m✖\u001b[39m \u001b[34mdplyr\u001b[39m::\u001b[32mcount()\u001b[39m           masks \u001b[34mmatrixStats\u001b[39m::count()\n","\u001b[31m✖\u001b[39m \u001b[34mtidyr\u001b[39m::\u001b[32mcrossing()\u001b[39m        masks \u001b[34migraph\u001b[39m::crossing()\n","\u001b[31m✖\u001b[39m \u001b[34mlubridate\u001b[39m::\u001b[32mdate()\u001b[39m        masks \u001b[34mbase\u001b[39m::date()\n","\u001b[31m✖\u001b[39m \u001b[34mdplyr\u001b[39m::\u001b[32mdesc()\u001b[39m            masks \u001b[34mIRanges\u001b[39m::desc()\n","\u001b[31m✖\u001b[39m \u001b[34mtidyr\u001b[39m::\u001b[32mexpand()\u001b[39m          masks \u001b[34mMatrix\u001b[39m::expand(), \u001b[34mS4Vectors\u001b[39m::expand()\n","\u001b[31m✖\u001b[39m \u001b[34mdplyr\u001b[39m::\u001b[32mfilter()\u001b[39m          masks \u001b[34mtidygraph\u001b[39m::filter(), \u001b[34mstats\u001b[39m::filter()\n","\u001b[31m✖\u001b[39m \u001b[34mdplyr\u001b[39m::\u001b[32mfirst()\u001b[39m           masks \u001b[34mS4Vectors\u001b[39m::first()\n","\u001b[31m✖\u001b[39m \u001b[34mdplyr\u001b[39m::\u001b[32mgroups()\u001b[39m          masks \u001b[34migraph\u001b[39m::groups(), \u001b[34mtidygraph\u001b[39m::groups()\n","\u001b[31m✖\u001b[39m \u001b[34mlubridate\u001b[39m::\u001b[32mintersect()\u001b[39m   masks \u001b[34mGenomicRanges\u001b[39m::intersect(), \u001b[34mGenomeInfoDb\u001b[39m::intersect(), \u001b[34mIRanges\u001b[39m::intersect(), \u001b[34mS4Vectors\u001b[39m::intersect(), \u001b[34mBiocGenerics\u001b[39m::intersect(), \u001b[34mbase\u001b[39m::intersect()\n","\u001b[31m✖\u001b[39m \u001b[34mdplyr\u001b[39m::\u001b[32mlag()\u001b[39m             masks \u001b[34mstats\u001b[39m::lag()\n","\u001b[31m✖\u001b[39m \u001b[34mtidyr\u001b[39m::\u001b[32mpack()\u001b[39m            masks \u001b[34mMatrix\u001b[39m::pack()\n","\u001b[31m✖\u001b[39m \u001b[34mggplot2\u001b[39m::\u001b[32mPosition()\u001b[39m      masks \u001b[34mBiocGenerics\u001b[39m::Position(), \u001b[34mbase\u001b[39m::Position()\n","\u001b[31m✖\u001b[39m \u001b[34mdplyr\u001b[39m::\u001b[32mrecode()\u001b[39m          masks \u001b[34mcar\u001b[39m::recode()\n","\u001b[31m✖\u001b[39m \u001b[34mpurrr\u001b[39m::\u001b[32mreduce()\u001b[39m          masks \u001b[34mGenomicRanges\u001b[39m::reduce(), \u001b[34mIRanges\u001b[39m::reduce()\n","\u001b[31m✖\u001b[39m \u001b[34mdplyr\u001b[39m::\u001b[32mrename()\u001b[39m          masks \u001b[34mtidygraph\u001b[39m::rename(), \u001b[34mS4Vectors\u001b[39m::rename()\n","\u001b[31m✖\u001b[39m \u001b[34mlubridate\u001b[39m::\u001b[32msecond()\u001b[39m      masks \u001b[34mS4Vectors\u001b[39m::second()\n","\u001b[31m✖\u001b[39m \u001b[34mlubridate\u001b[39m::\u001b[32msecond<-()\u001b[39m    masks \u001b[34mS4Vectors\u001b[39m::second<-()\n","\u001b[31m✖\u001b[39m \u001b[34mlubridate\u001b[39m::\u001b[32msetdiff()\u001b[39m     masks \u001b[34mGenomicRanges\u001b[39m::setdiff(), \u001b[34mIRanges\u001b[39m::setdiff(), \u001b[34mS4Vectors\u001b[39m::setdiff(), \u001b[34mBiocGenerics\u001b[39m::setdiff(), \u001b[34mbase\u001b[39m::setdiff()\n","\u001b[31m✖\u001b[39m \u001b[34mpurrr\u001b[39m::\u001b[32msimplify()\u001b[39m        masks \u001b[34migraph\u001b[39m::simplify()\n","\u001b[31m✖\u001b[39m \u001b[34mdplyr\u001b[39m::\u001b[32mslice()\u001b[39m           masks \u001b[34mtidygraph\u001b[39m::slice(), \u001b[34mIRanges\u001b[39m::slice()\n","\u001b[31m✖\u001b[39m \u001b[34mpurrr\u001b[39m::\u001b[32msome()\u001b[39m            masks \u001b[34mcar\u001b[39m::some()\n","\u001b[31m✖\u001b[39m \u001b[34mreadr\u001b[39m::\u001b[32mspec()\u001b[39m            masks \u001b[34mgenefilter\u001b[39m::spec()\n","\u001b[31m✖\u001b[39m \u001b[34mlubridate\u001b[39m::\u001b[32mstamp()\u001b[39m       masks \u001b[34mcowplot\u001b[39m::stamp()\n","\u001b[31m✖\u001b[39m \u001b[34mlubridate\u001b[39m::\u001b[32munion()\u001b[39m       masks \u001b[34migraph\u001b[39m::union(), \u001b[34mGenomicRanges\u001b[39m::union(), \u001b[34mIRanges\u001b[39m::union(), \u001b[34mS4Vectors\u001b[39m::union(), \u001b[34mBiocGenerics\u001b[39m::union(), \u001b[34mbase\u001b[39m::union()\n","\u001b[31m✖\u001b[39m \u001b[34mtidyr\u001b[39m::\u001b[32munpack()\u001b[39m          masks \u001b[34mMatrix\u001b[39m::unpack()\n","\n"],"name":"stderr"}]},{"cell_type":"code","metadata":{"id":"fYOfc-ZDTQBC","executionInfo":{"status":"ok","timestamp":1627462796451,"user_tz":-480,"elapsed":46940,"user":{"displayName":"Indrik Wijaya","photoUrl":"","userId":"00208067985282844199"}}},"source":["methPath <- file.path(tempdir(),\"meth.RData\")\n","if(!file.exists(methPath)) {\n","  download.file(\"https://www.huber.embl.de/users/jlu/data/meth.RData\",\n","                methPath)\n","  load(methPath)\n","} else {\n","  load(methPath)\n","}"],"execution_count":17,"outputs":[]},{"cell_type":"code","metadata":{"id":"fsm6mxqEVstb","executionInfo":{"status":"ok","timestamp":1627462797938,"user_tz":-480,"elapsed":401,"user":{"displayName":"Indrik Wijaya","photoUrl":"","userId":"00208067985282844199"}}},"source":["load('drug.RData')\n","load('gene.RData')\n","load('rna.RData')"],"execution_count":18,"outputs":[]},{"cell_type":"code","metadata":{"id":"Tq80S73lV330","executionInfo":{"status":"ok","timestamp":1627462847721,"user_tz":-480,"elapsed":280,"user":{"displayName":"Indrik Wijaya","photoUrl":"","userId":"00208067985282844199"}}},"source":["viabMat <- mutate(drug, id = paste0(Drug, '_', concIndex)) %>%\n","  select(patientID, id, normVal) %>%\n","  spread(key = patientID, value = normVal) %>%\n","  data.frame() %>% column_to_rownames('id') %>%\n","  as.matrix()"],"execution_count":19,"outputs":[]},{"cell_type":"code","metadata":{"id":"ISjhalyAWEnk","executionInfo":{"status":"ok","timestamp":1627462932364,"user_tz":-480,"elapsed":25783,"user":{"displayName":"Indrik Wijaya","photoUrl":"","userId":"00208067985282844199"}}},"source":["rna.vst<-DESeq2::varianceStabilizingTransformation(rna)\n","exprMat <- assay(rna.vst)\n","nTop = 5000\n","sds <- genefilter::rowSds(exprMat)\n","exprMat <- exprMat[order(sds, decreasing = T)[1:nTop],]"],"execution_count":20,"outputs":[]},{"cell_type":"code","metadata":{"id":"A4G3n1-4WTDU","executionInfo":{"status":"ok","timestamp":1627462940769,"user_tz":-480,"elapsed":8411,"user":{"displayName":"Indrik Wijaya","photoUrl":"","userId":"00208067985282844199"}}},"source":["methData <- assays(meth)[[\"beta\"]]\n","nTop = 5000\n","sds <- genefilter::rowSds(methData)\n","methData <- methData[order(sds, decreasing = T)[1:nTop],]"],"execution_count":21,"outputs":[]},{"cell_type":"code","metadata":{"id":"Suny22PtWUan","executionInfo":{"status":"ok","timestamp":1627462941815,"user_tz":-480,"elapsed":9,"user":{"displayName":"Indrik Wijaya","photoUrl":"","userId":"00208067985282844199"}}},"source":["gene <- t(gene)"],"execution_count":22,"outputs":[]},{"cell_type":"code","metadata":{"id":"534pkkkMWYN8","executionInfo":{"status":"ok","timestamp":1627462942684,"user_tz":-480,"elapsed":877,"user":{"displayName":"Indrik Wijaya","photoUrl":"","userId":"00208067985282844199"}}},"source":["mofaData <- list(Drugs = viabMat, \n","                mRNA = exprMat,\n","                Mutations = gene, \n","                Methylation = methData)\n","\n","# Create MultiAssayExperiment object \n","mofaData <- MultiAssayExperiment::MultiAssayExperiment(\n","  experiments = mofaData\n",")"],"execution_count":23,"outputs":[]},{"cell_type":"code","metadata":{"id":"TInpXo1tWZ6t","executionInfo":{"status":"ok","timestamp":1627462944270,"user_tz":-480,"elapsed":1591,"user":{"displayName":"Indrik Wijaya","photoUrl":"","userId":"00208067985282844199"}}},"source":["useSamples <- MultiAssayExperiment::sampleMap(mofaData) %>%\n","  as_tibble() %>% group_by(primary) %>% summarise(n= length(assay)) %>%\n","  filter(n >= 3) %>% pull(primary)\n","mofaData <- mofaData[,useSamples]"],"execution_count":24,"outputs":[]},{"cell_type":"code","metadata":{"colab":{"base_uri":"https://localhost:8080/","height":98},"id":"YVARm4i3WbkW","executionInfo":{"status":"ok","timestamp":1627462951835,"user_tz":-480,"elapsed":350,"user":{"displayName":"Indrik Wijaya","photoUrl":"","userId":"00208067985282844199"}},"outputId":"6937ed9f-352c-43fb-d9f9-e7a3e0b87487"},"source":["experiments(mofaData)"],"execution_count":25,"outputs":[{"output_type":"display_data","data":{"text/plain":["ExperimentList class object of length 4:\n"," [1] Drugs: matrix with 315 rows and 190 columns\n"," [2] mRNA: matrix with 5000 rows and 202 columns\n"," [3] Mutations: matrix with 39 rows and 217 columns\n"," [4] Methylation: matrix with 5000 rows and 158 columns"]},"metadata":{"tags":[]}}]},{"cell_type":"code","metadata":{"colab":{"base_uri":"https://localhost:8080/","height":65},"id":"Y6KfnlCxWeBO","executionInfo":{"status":"ok","timestamp":1627462965841,"user_tz":-480,"elapsed":338,"user":{"displayName":"Indrik Wijaya","photoUrl":"","userId":"00208067985282844199"}},"outputId":"9177954c-a0bf-45f2-bf21-e233d425978e"},"source":["table(table(sampleMap(mofaData)$primary))"],"execution_count":26,"outputs":[{"output_type":"display_data","data":{"text/plain":["\n","  3   4 \n","101 116 "]},"metadata":{"tags":[]}}]},{"cell_type":"code","metadata":{"colab":{"base_uri":"https://localhost:8080/","height":131},"id":"bBcrEw-tWhcg","executionInfo":{"status":"ok","timestamp":1627463036108,"user_tz":-480,"elapsed":298,"user":{"displayName":"Indrik Wijaya","photoUrl":"","userId":"00208067985282844199"}},"outputId":"5f0f4ec6-381f-4512-f87f-c3d9ca2063cb"},"source":["MOFAobject <- createMOFAobject(mofaData)\n","MOFAobject"],"execution_count":27,"outputs":[{"output_type":"stream","text":["Creating MOFA object from a MultiAssayExperiment object...\n","\n"],"name":"stderr"},{"output_type":"display_data","data":{"text/plain":["Untrained MOFA model with the following characteristics: \n","  Number of views: 4 \n","  View names: Drugs mRNA Mutations Methylation \n","  Number of features per view: 315 5000 39 5000\n","  Number of samples: 217 "]},"metadata":{"tags":[]}}]},{"cell_type":"code","metadata":{"colab":{"base_uri":"https://localhost:8080/","height":92},"id":"LUcF9TEkWynH","executionInfo":{"status":"error","timestamp":1627463048626,"user_tz":-480,"elapsed":281,"user":{"displayName":"Indrik Wijaya","photoUrl":"","userId":"00208067985282844199"}},"outputId":"d523ca1b-f299-4a06-d708-a859556bbbb0"},"source":["p <- plotDataOverview.m(MOFAobject)\n","p$plot"],"execution_count":28,"outputs":[{"output_type":"error","ename":"ERROR","evalue":"ignored","traceback":["Error in plotDataOverview.m(MOFAobject): could not find function \"plotDataOverview.m\"\nTraceback:\n"]}]},{"cell_type":"code","metadata":{"id":"Tw4Cb4fYW1qy","executionInfo":{"status":"ok","timestamp":1627463118230,"user_tz":-480,"elapsed":337,"user":{"displayName":"Indrik Wijaya","photoUrl":"","userId":"00208067985282844199"}}},"source":["DataOptions <- getDefaultDataOptions()\n"],"execution_count":29,"outputs":[]},{"cell_type":"code","metadata":{"colab":{"base_uri":"https://localhost:8080/","height":143},"id":"wmBf_wbCXGpX","executionInfo":{"status":"ok","timestamp":1627463124626,"user_tz":-480,"elapsed":324,"user":{"displayName":"Indrik Wijaya","photoUrl":"","userId":"00208067985282844199"}},"outputId":"5ef40b6b-6ad3-4b67-cb59-12dfea01ed80"},"source":["ModelOptions <- getDefaultModelOptions(MOFAobject)\n","ModelOptions$numFactors <- 30\n","ModelOptions$sparsity <- TRUE\n","ModelOptions"],"execution_count":30,"outputs":[{"output_type":"display_data","data":{"text/plain":["$likelihood\n","      Drugs        mRNA   Mutations Methylation \n"," \"gaussian\"  \"gaussian\" \"bernoulli\"  \"gaussian\" \n","\n","$numFactors\n","[1] 30\n","\n","$sparsity\n","[1] TRUE\n"],"text/latex":"\\begin{description}\n\\item[\\$likelihood] \\begin{description*}\n\\item[Drugs] 'gaussian'\n\\item[mRNA] 'gaussian'\n\\item[Mutations] 'bernoulli'\n\\item[Methylation] 'gaussian'\n\\end{description*}\n\n\\item[\\$numFactors] 30\n\\item[\\$sparsity] TRUE\n\\end{description}\n","text/markdown":"$likelihood\n:   Drugs\n:   'gaussian'mRNA\n:   'gaussian'Mutations\n:   'bernoulli'Methylation\n:   'gaussian'\n\n\n$numFactors\n:   30\n$sparsity\n:   TRUE\n\n\n","text/html":["<dl>\n","\t<dt>$likelihood</dt>\n","\t\t<dd><style>\n",".dl-inline {width: auto; margin:0; padding: 0}\n",".dl-inline>dt, .dl-inline>dd {float: none; width: auto; display: inline-block}\n",".dl-inline>dt::after {content: \":\\0020\"; padding-right: .5ex}\n",".dl-inline>dt:not(:first-of-type) {padding-left: .5ex}\n","</style><dl class=dl-inline><dt>Drugs</dt><dd>'gaussian'</dd><dt>mRNA</dt><dd>'gaussian'</dd><dt>Mutations</dt><dd>'bernoulli'</dd><dt>Methylation</dt><dd>'gaussian'</dd></dl>\n","</dd>\n","\t<dt>$numFactors</dt>\n","\t\t<dd>30</dd>\n","\t<dt>$sparsity</dt>\n","\t\t<dd>TRUE</dd>\n","</dl>\n"]},"metadata":{"tags":[]}}]},{"cell_type":"code","metadata":{"colab":{"base_uri":"https://localhost:8080/","height":209},"id":"bouA1OkYXILL","executionInfo":{"status":"ok","timestamp":1627463132553,"user_tz":-480,"elapsed":299,"user":{"displayName":"Indrik Wijaya","photoUrl":"","userId":"00208067985282844199"}},"outputId":"c4d49fab-3022-41d6-8dde-acd84afac313"},"source":["TrainOptions <- getDefaultTrainOptions()\n","\n","# Automatically drop factors that explain less than 2% of variance in all omics\n","TrainOptions$DropFactorThreshold <- 0.02\n","TrainOptions$tolerance <- 0.01\n","TrainOptions"],"execution_count":31,"outputs":[{"output_type":"display_data","data":{"text/plain":["$maxiter\n","[1] 2000\n","\n","$tolerance\n","[1] 0.01\n","\n","$DropFactorThreshold\n","[1] 0.02\n","\n","$verbose\n","[1] FALSE\n","\n","$seed\n","NULL\n"],"text/latex":"\\begin{description}\n\\item[\\$maxiter] 2000\n\\item[\\$tolerance] 0.01\n\\item[\\$DropFactorThreshold] 0.02\n\\item[\\$verbose] FALSE\n\\item[\\$seed] NULL\n\\end{description}\n","text/markdown":"$maxiter\n:   2000\n$tolerance\n:   0.01\n$DropFactorThreshold\n:   0.02\n$verbose\n:   FALSE\n$seed\n:   NULL\n\n\n","text/html":["<dl>\n","\t<dt>$maxiter</dt>\n","\t\t<dd>2000</dd>\n","\t<dt>$tolerance</dt>\n","\t\t<dd>0.01</dd>\n","\t<dt>$DropFactorThreshold</dt>\n","\t\t<dd>0.02</dd>\n","\t<dt>$verbose</dt>\n","\t\t<dd>FALSE</dd>\n","\t<dt>$seed</dt>\n","\t\t<dd>NULL</dd>\n","</dl>\n"]},"metadata":{"tags":[]}}]},{"cell_type":"code","metadata":{"colab":{"base_uri":"https://localhost:8080/","height":224},"id":"K-1Z606dXZ4A","executionInfo":{"status":"error","timestamp":1627463202315,"user_tz":-480,"elapsed":2132,"user":{"displayName":"Indrik Wijaya","photoUrl":"","userId":"00208067985282844199"}},"outputId":"4a63c50b-1d2b-4429-ee5f-9820bab3477d"},"source":["library(reticulate)\n","py_install(\"mofapy\", envname = \"r-reticulate\", method=\"auto\")"],"execution_count":33,"outputs":[{"output_type":"stream","text":["Using Python: /usr/bin/python3.7\n","Creating virtual environment 'r-reticulate' ... FAILED\n"],"name":"stdout"},{"output_type":"error","ename":"ERROR","evalue":"ignored","traceback":["Error: Error creating virtual environment 'r-reticulate' [error code 1]\nTraceback:\n","1. py_install(\"mofapy\", envname = \"r-reticulate\", method = \"auto\")","2. virtualenv_install(envname = envname, packages = packages, ...)","3. virtualenv_create(envname)","4. stopf(fmt, name, result)","5. stop(sprintf(fmt, ...), call. = call.)"]}]},{"cell_type":"code","metadata":{"colab":{"base_uri":"https://localhost:8080/","height":455},"id":"TD9XjWSbXKJy","executionInfo":{"status":"error","timestamp":1627463150278,"user_tz":-480,"elapsed":3463,"user":{"displayName":"Indrik Wijaya","photoUrl":"","userId":"00208067985282844199"}},"outputId":"8e33aa47-3d70-41bc-dcf4-c825a0ae9f48"},"source":["MOFAobject <- prepareMOFA(\n","  MOFAobject, \n","  DataOptions = DataOptions,\n","  ModelOptions = ModelOptions,\n","  TrainOptions = TrainOptions\n",")\n","\n","MOFAobject <- runMOFA(MOFAobject)\n"],"execution_count":32,"outputs":[{"output_type":"stream","text":["Checking data options...\n","\n","Checking training options...\n","\n","Checking model options...\n","\n"],"name":"stderr"},{"output_type":"stream","text":["[1] \"No output file provided, using a temporary file...\"\n"],"name":"stdout"},{"output_type":"error","ename":"ERROR","evalue":"ignored","traceback":["Error in value[[3L]](cond): mofapy package not found. Make sure that Reticulate is pointing to the right Python binary.\n         Please read the instructions here: https://github.com/bioFAM/MOFA#installation\nTraceback:\n","1. runMOFA(MOFAobject)","2. tryCatch({\n .     import(\"mofapy\")\n . }, error = function(err) {\n .     stop(\"mofapy package not found. Make sure that Reticulate is pointing to the right Python binary.\\n         Please read the instructions here: https://github.com/bioFAM/MOFA#installation\")\n . })","3. tryCatchList(expr, classes, parentenv, handlers)","4. tryCatchOne(expr, names, parentenv, handlers[[1L]])","5. value[[3L]](cond)","6. stop(\"mofapy package not found. Make sure that Reticulate is pointing to the right Python binary.\\n         Please read the instructions here: https://github.com/bioFAM/MOFA#installation\")"]}]},{"cell_type":"code","metadata":{"id":"RBJXAVr6XNtG"},"source":[""],"execution_count":null,"outputs":[]}]}